server_list/tasks.py

```diff
@@ -0,0 +1,100 @@
+import asyncio
+import json
+import os
+from datetime import datetime
+
+from .app import app, celery, db
+from .models import Server, Stats
+from .ping import ping_servers_async, ping_server_addresses
+from .util import get_geo_continent, server_ranking
+
+
+@celery.task
+def update_server(obj):
+	geo_continent = get_geo_continent(obj["ip"])
+	if geo_continent is not None:
+		obj["geo_continent"] = geo_continent
+
+	# Ensure that a Minetest server is actually reachable on all addresses
+	pings = ping_server_addresses(obj["address"], obj["port"])
+
+	if pings is None:
+		return
+
+	# Use average ping
```

Comment:
So the final ping we calculate is this, correct?
<pre>
average of(
  minimum of(
    ping(address) three times
  ) for each address
)

Reply:
Yes.